cmd/bootstrap/cmd/access_keygen.go

```diff
@@ -0,0 +1,145 @@
+
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/model/bootstrap"
+	"github.com/onflow/flow-go/utils/grpcutils"
+	"github.com/onflow/flow-go/utils/io"
+)
+
+const certValidityPeriod = 100 * 365 * 24 * time.Hour // ~100 years
+
+var (
+	flagCommonName     string
+	flagNodeInfoFile   string
+	flagOutputKeyFile  string
+	flagOutputCertFile string
+)
+
+var accessKeyCmd = &cobra.Command{
+	Use:   "access-keygen",
+	Short: "Generate access node grpc TLS key and certificate",
+	Run:   accessKeyCmdRun,
+}
+
+func init() {
+	rootCmd.AddCommand(accessKeyCmd)
+
+	accessKeyCmd.Flags().StringVar(&flagNodeInfoFile, "node-info", "", "path to node's node-info.priv.json file")
+	_ = accessKeyCmd.MarkFlagRequired("node-info")
+
+	accessKeyCmd.Flags().StringVar(&flagOutputKeyFile, "key", "./access-tls.key", "path to output private key file")
+	accessKeyCmd.Flags().StringVar(&flagOutputCertFile, "cert", "./access-tls.crt", "path to output certificate file")
+	accessKeyCmd.Flags().StringVar(&flagCommonName, "cn", "", "common name to include in the certificate")
+}
+
+// accessKeyCmdRun generate an Access node TLS key and certificate
+func accessKeyCmdRun(_ *cobra.Command, _ []string) {
+	networkKey, err := loadNetworkKey(flagNodeInfoFile)
+	if err != nil {
+		log.Fatal().Msgf("could not load node-info file: %v", err)
+	}
+
+	certTmpl, err := defaultCertTemplate()
+	if err != nil {
+		log.Fatal().Msgf("could not create certificate template: %v", err)
+	}
+
+	if flagCommonName != "" {
+		log.Info().Msgf("using cn: %s", flagCommonName)
+		// needed to pass envoy's certificate validity check
+		certTmpl.Subject.CommonName = flagCommonName
+	}
+
+	cert, err := grpcutils.X509Certificate(networkKey, grpcutils.WithCertTemplate(certTmpl))
+	if err != nil {
+		log.Fatal().Msgf("could not generate key pair: %v", err)
+	}
+
+	// write cert and private key to disk
+	keyBytes, err := x509.MarshalECPrivateKey(cert.PrivateKey.(*ecdsa.PrivateKey))
+	if err != nil {
+		log.Fatal().Msgf("could not encode private key: %v", err)
+	}
+
+	err = writeFile(flagOutputKeyFile, pem.EncodeToMemory(&pem.Block{
+		Type:  "EC PRIVATE KEY",
+		Bytes: keyBytes,
+	}))
+	if err != nil {
+		log.Fatal().Msgf("could not write private key: %v", err)
+	}
+
+	err = writeFile(flagOutputCertFile, pem.EncodeToMemory(&pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: cert.Certificate[0],
+	}))
+	if err != nil {
+		log.Fatal().Msgf("could not write certificate: %v", err)
+	}
+}
+
+func loadNetworkKey(nodeInfoPath string) (crypto.PrivateKey, error) {
+	data, err := io.ReadFile(nodeInfoPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not read private node info (path=%s): %w", nodeInfoPath, err)
+	}
+
+	var info bootstrap.NodeInfoPriv
+	err = json.Unmarshal(data, &info)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse private node info (path=%s): %w", nodeInfoPath, err)
+	}
+
+	return info.NetworkPrivKey.PrivateKey, nil
+}
+
+func writeFile(path string, data []byte) error {
```

Comment:
nit: there is also a `io.WriteFile` or `os.WriteFile`

Reply:
465383f